File: 78.py
class Solution:
    def subsets(self, nums: list[int]) -> list[list[int]]:
        res = [[]]
        com = []

        from itertools import combinations
        for i in range(len(nums)):
            tem = combinations(nums, i+1)
            com.append(tem)

        for i in com:
            for j in i:
                res.append(list(j))

        return res


print(Solution().subsets([1,2,3]))

File: 78.go
func subsets(nums []int) [][]int {
	var subsets [][]int
	var curSubset []int
	FindAllSubsets(0, nums, &subsets, curSubset)
	return subsets
}

func FindAllSubsets(ind int, nums []int, subsets *[][]int, curSubset []int) {
	if ind >= len(nums) {
		*subsets = append(*subsets, append([]int{}, curSubset...))
		return
	}

	curSubset = append(curSubset, nums[ind])
	FindAllSubsets(ind+1, nums, subsets, curSubset)

	curSubset = curSubset[:len(curSubset)-1]
	FindAllSubsets(ind+1, nums, subsets, curSubset)
}